7_react/react1/week2/todo-app/src/components/TodoItem/TodoItem.jsx

```diff
@@ -0,0 +1,41 @@
+import PropTypes from 'prop-types';
+import Task from '../Task/Task';
+import Deadline from '../Deadline/Deadline';
+import { useState } from 'react';
+import './TodoItem.css';
+function TodoItem({ todo, removeTodo }) {
+  const [isChecked, setIsChecked] = useState(false);
+
+  const handleChecboxChange = () => {
+    setIsChecked(!isChecked);
+  }
+
+  const handleRemoveClick = () => {
+    removeTodo(todo.id);
+  };
+  
+
+  return (
+    <div className={`todo-item ${isChecked ? 'completed' : ''}`}>
+      <input
+        type="checkbox"
+        checked={isChecked}
+        onChange={handleChecboxChange}
+      />
+      <Task task={todo.task} isChecked={isChecked}/>
+      <Deadline deadline={todo.deadline} />
+      <button className="remove-button" onClick={handleRemoveClick}>x</button>
+    </div>
+  );
+}
+
+TodoItem.propTypes = {
```

Comment:
Very nice to set some types and the like so we can catch pesky type errors and missing props! 👍 `Deadline`, `Task`, and `TodoList` are missing them though. They are not required, but consistency is key! 😄

Reply:
Thank you🙏 I just used PropTypes for `Deadline`, `Task` and `TodoList`.